dashboard/src/shared/I18n.ts

```diff
@@ -2,44 +2,47 @@ import messages_en from "../locales/en.json";
 import { axiosWithAuth } from "./AxiosInstance";
 
 export interface II18nConfig {
-  locale: string;
+  locale: ISupportedLangs | "custom";
   messages: Record<string, string>;
 }
 
+// Add here new new supported languages literals
 export enum ISupportedLangs {
   en = "en",
-  custom = "custom",
 }
 
 const messages = {};
-messages[ISupportedLangs.en] = messages_en;
 
-export function getDefaulI18nConfig(): II18nConfig {
-  return { locale: ISupportedLangs.en, messages: messages[ISupportedLangs.en] };
-}
+// Load here the compiled messages for each supported language
+messages[ISupportedLangs.en] = messages_en;
 
-export function getI18nConfig(lang: ISupportedLangs): II18nConfig {
-  if (lang && ISupportedLangs[lang]) {
-    return { locale: lang, messages: messages[lang] };
-  } else {
-    return getDefaulI18nConfig();
+export default class I18n {
+  public static getDefaulI18nConfig(): II18nConfig {
```

Comment:
I am not sure why you have moved these to a class but if it's already named `I18n` you can remove that from the method names `I18n.getDefaultConfig()`...